src/meetingsession/DefaultMeetingSession.ts

```diff
@@ -207,4 +211,46 @@ export default class DefaultMeetingSession implements MeetingSession, Destroyabl
       }
     }
   }
+
+  private addEventlistenerForCSPViolations(): void {
+    /* istanbul ignore else */
+    if (
+      !('window' in global) ||
+      !window.addEventListener ||
+      !('document' in global) ||
+      !document.addEventListener
+    ) {
+      return;
+    }
+    /* istanbul ignore next */
+    document.addEventListener('securitypolicyviolation', (e: SecurityPolicyViolationEvent) => {
+      this.logger.error(
+        'Security Policy Violation\n' +
+          `Blocked uri: ${e.blockedURI}\n` +
+          `Violated directive: ${e.violatedDirective}\n` +
+          `Original policy: ${e.originalPolicy}\n` +
+          `Document URI: ${e.documentURI}\n` +
+          `Source file: ${e.sourceFile}\n` +
+          `Line No.: ${e.lineNumber}\n`
+      );
+    });
+  }
+
+  private removeCSPViolationEventListeners(): void {
+    /* istanbul ignore else */
+    if (
+      !('window' in global) ||
+      !window.removeEventListener ||
+      !('document' in global) ||
+      !document.removeEventListener
+    ) {
+      return;
+    }
+    /* istanbul ignore next */
+    window.removeEventListener('securitypolicyviolation', () => {
```

Comment:
You must pass the same event listener function with `addEventListener`. One option is to return an unsbuscriber from the subscribe function.

```js
private subscribeToCSPViolationEvent(): () => void {
  const listener = () => { /* ... */ };

  document.addEventListener('securitypolicyviolation', listener);
  return () => {
     document.removeEventListener('securitypolicyviolation', listener);
  };
}
```

```js
this.unsubscribeFromCSPViolationEvent = this.subscribeToCSPViolationEvent();

async destroy(): Promise<void> {
  // ...
  this.unsubscribeFromCSPViolationEvent?.();
}
```

Reply:
Good point. Will do.